src/errors.rs

```diff
@@ -491,6 +495,31 @@ fn component_unavailable_msg(cs: &[Component], manifest: &Manifest, toolchain: &
     String::from_utf8(buf).unwrap()
 }
 
+fn components_missing_msg(cs: &[Component], manifest: &Manifest, toolchain: &str) -> String {
+    assert!(!cs.is_empty());
+    let mut buf = vec![];
+
+    let _ = writeln!(
+        buf,
+        "component {} is unavailable for download for channel '{}'",
+        &cs[0].description(manifest),
```

Comment:
Why only report the first missing component?

Reply:
Fixed.